src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -304,27 +283,45 @@ private RestClientMethod BuildMethod(Operation operation, string clientName, IRe
             );
 
             ResponseBody? responseBody = null;
-            if (response is SchemaResponse schemaResponse && httpResponse != null)
+            ResponseHeaderGroupType? responseHeaderModel = null;
+            string operationName = operation.CSharpName();
+
+            //TODO: Handle multiple responses: https://github.com/Azure/autorest.csharp/issues/413
+            ServiceResponse? response = null;
+
+            // if operation is a long-running operation than we're generating an initial call here so find a response with non 200/204 code
+            // fallback to the first on otherwise
+
+            if (operation.IsLongRunning)
             {
-                Schema schema = schemaResponse.Schema is ConstantSchema constantSchema ? constantSchema.ValueType : schemaResponse.Schema;
-                CSharpType responseType = _typeFactory.CreateType(schema, isNullable: false);
+                response = operation.LongRunningInitialResponse;
+            }
 
-                ObjectSerialization serialization = _serializationBuilder.Build(httpResponse.KnownMediaType, schema, isNullable: false);
+            response ??= operation.Responses.FirstOrDefault();
 
-                responseBody = new ObjectResponseBody(responseType, serialization);
+            Response clientResponse;
+            if (response != null)
+            {
+                // Ignore response body and headers for LROs as the ArmOperationHelpers figures out them dynamically
+                if (!operation.IsLongRunning)
+                {
+                    responseBody = BuildResponseBody(response);
+
+                    responseHeaderModel = BuildResponseHeaderModel(operation, response);
+                }
+
+                clientResponse = new Response(
+                    responseBody,
+                    response.HttpResponse.StatusCodes.Select(ToStatusCode).ToArray(),
+                    responseHeaderModel
+                );
             }
-            else if (response is BinaryResponse)
+            else
             {
-                responseBody = new StreamResponseBody();
+                // Special case for httpInfrastructure testServer swagger, service method always fails
+                clientResponse = new Response(null, Array.Empty<int>(), null);
```

Comment:
Is this case supposed to pass for that test? I'm confused.

Reply:
This is only to make `httpInfrastructure` weird test swagger work. Nothing else should really be hitting this.

Reply:
I would prefer to have an exception where there is no response but can't.

Reply:
So, the generator fails for `httpInfrastructure` if that isn't there? What changed from before where this wasn't needed?

Reply:
`clientResponse` was nullable and we checked it all the time.